### ISSUE TITLE
the trigger doesn't halt the process although the exit status is 1

### ISSUE THREAD

**Issue by lzecca78** [hashicorp/vagrant#9992]
### Vagrant version
vagrant  2.1.2

### Host operating system
Happens in both Mac (High Sierra) and Linux (ubuntu 16.04)

### Guest operating system
Ubuntu 16.04

### Vagrantfile
```ruby
  config.trigger.before :up, :provision do |trigger|
    trigger.name = "upgrade or install librarian-puppet libs"
    trigger.info = "upgrade or install librarian-puppet libs"
    trigger.on_error = :halt
    trigger.run  = {path: "bin/librarian-puppet.sh"}
  end

```
the `bin/libriarian-puppet.sh` is : 

``` shell
#!/bin/bash
set -e errexit
set -o pipefail

echo "removing librarian dotfiles..."
cd environments/production && rm -rf .librarian && cd -
echo "installing modules with Librarian-Puppet, this could take a while due to network downloads...."
cd environments/production && librarian-puppet install --verbose
```

### Debug output
https://gist.github.com/lzecca78/7fd46d2b11c5a8427efc6c28f90607af

### Expected behavior
Due to the fact that the script exit with status 1, the expected behavior is that Vagrant stop the process suddenly

### Actual behavior
Although the script exit with status 1, the provision flow goes on 

### Steps to reproduce
1. Just add the trigger to a plain Vagrantfile
2. Create a Puppetfile with a non-existing version of a puppet module
3. Launch vagrant up

**Comment by briancain** [comment]
Related to #9997

**Comment by briancain** [comment]
Closed via #10005

**Comment by lzecca78** [comment]
thank you @briancain  for fixing the issue! Do you know when this fix will be released?

**Comment by briancain** [comment]
@lzecca78 - We are planning for this fix to go out in the `2.1.3` release and that's expected to go out in the next few weeks!
